### PR TITLE
Intentionally forget start_args when job is done

### PR DESCRIPTION
**Pull request by AlanCoding**
Connect https://github.com/ansible/awx/issues/742

### CODE REVIEW

**Comment by AlanCoding** [comment]
this missed marking job as error

**Comment by AlanCoding** [comment]
A short blurb on comprehensiveness:

Out of all of the unified job types, `job` is the only one that can accept passwords on launch.

That means that we only need to concern ourselves with the job status transitions. Out of all job status options, there are a limited number of "terminal" options:

 - successful
 - failed
 - error
 - canceled

By "terminal", I mean that once a job is in this state, this particular job can never ever run to completion. In all cases, we are "done" with the start args, and they should be trashed.

This PR blanks start args in 3 different cases.

The case of `canceled` can only be hit within a single code-route. We can only cancel jobs by calling the cancel method to set `cancel_flag`, and the cancel method will always blank start args.

Blanking start args at the start of running the job takes care of a myriad of scenarios for the transition to `successful`, `failed`, and `error`, and may be benignly duplicate with the cancel method.

The case of `error` can be hit by several mechanisms, but outside of the job celery task itself, AFAIK, the only way to transition to error is by the task manager's reaping code.

**Review by matburt — APPROVED** [review]

File: awx/main/migrations/_reencrypt.py [hunk 1]
```diff
@@ -74,3 +74,20 @@ def _unified_jobs(apps):
                 uj.start_args = decrypt_field(uj, 'start_args')
                 uj.start_args = encrypt_field(uj, 'start_args')
                 uj.save()
+
+
+def blank_old_start_args(apps, schema_editor):
+    UnifiedJob = apps.get_model('main', 'UnifiedJob')
+    for uj in UnifiedJob.objects.defer('result_stdout_text').exclude(start_args='').iterator():
```
Comment (ryanpetrello) [inline awx/main/migrations/_reencrypt.py:81]: Yay for `defer()` 😄

**Review by ryanpetrello — APPROVED** [review]
Sorry this one took me so long to look to stare at, looks good @AlanCoding

**Pull request merged**